src/pkg/oidc/secret.go

```diff
@@ -86,6 +87,9 @@ var m SecretManager = &defaultManager{
 func (dm *defaultManager) VerifySecret(ctx context.Context, username string, secret string) (*UserInfo, error) {
 	log.Debugf("Verifying the secret for user: %s", username)
 	oidcUser, err := dm.metaDao.GetByUsername(ctx, username)
+	if strings.Contains(err.Error(), "no row found") {
```

Comment:
Checking error strings is fragile. Would not be better to:
1.  Use proper error types or sentinel errors
2. Check if the error implements a specific interface